src/main/java/com/spartronics4915/frc2020/subsystems/LED.java

```diff
@@ -0,0 +1,100 @@
+package com.spartronics4915.frc2020.subsystems;
+
+import com.spartronics4915.lib.util.Logger;
+import com.spartronics4915.lib.subsystems.SpartronicsSubsystem;
+
+import edu.wpi.first.wpilibj.SerialPort;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ * The subsystem that controls the LED.
+ * Communicating to the Arduino by giving
+ * a byte using Serial to the wanted BlingState to be
+ * read by the Arduino to show the given Bling style wanted.
+ *
+ * Based off the Arduino code under Spartronics4915/Bling/Roborio_functionality
+ */
+public class LED extends SpartronicsSubsystem
+{
+    private SerialPort mBling;
+
+    private static LED sInstance = null;
+    private static BlingState mBlingState;
+
+    public static LED getInstance()
+    {
+        if (sInstance == null)
+        {
+            sInstance = new LED();
+            mBlingState = BlingState.OFF;
+        }
+        return sInstance;
+    }
+
+    private LED()
+    {
+        try
+        {
+            mBling = new SerialPort(9600, SerialPort.Port.kUSB);
+
+            logInitialized(true);
+        }
+        catch (Exception e)
+        {
+            logException("LED: Couldn't initialize SerialPort", e);
+            logInitialized(false);
+        }
+    }
+
+    // TODO: validate desired periodic() function for LED subsystem
+    @Override
+    public void periodic()
+    {
+        SmartDashboard.putString("LED state:", mBlingState.toString());
+    }
+
+    /**
+     * This enum is giving the possible styles we can have the Arduino express.
+     */
+    public enum BlingState
+    {
+        // TODO: Add blingStates & ensure it matches Arduino sketch code
+        // bling code is passed in for use in the BlingState methods
+        OFF("0"), // turn off bling
+        DISABLED("1"), // robot powered on but disabled (in disabledInit())
+        SOME_BLING_STATE_2("2"), // ...
+        SOME_BLING_STATE_3("3"), // ...
+        ; // semicolon to state more to follow
+
+        private final String blingCode;
+
+        BlingState(String code)
+        {
+            this.blingCode = code;
+        }
+
+        public byte[] getBlingMessage()
+        {
+            return this.blingCode.getBytes();
+        }
+    }
+
+    /**
+     * This will go through what we want the bling to do and express that style of bling.
+     */
+    public void setBlingState(BlingState blingState)
+    {
+        if (!isInitialized())
```

Comment:
This check could be done after storing and logging the blingState. It only needs to come before the mBling.write() call. If we allow the other stuff to happen, then the logfiles will be consistent, and the programmers can get some feedback even without an Arduino attached. It might be nice to not in the log message that the system was not initialized, to help in debugging.

Reply:
adding a todo